CMSgov/design-system
# [WNMGDS-2785] Fix first batch of Figma/CSS differences

## Summary

### BREAKING changes

- Removed `--button_font-weight` variable in favor of the following new variables:
    - `--font-weight-button-sm`
    - `--font-weight-button-md`
    - `--font-weight-button-lg`
- Removed these font variables:
    - `--font-sans`
    - `--font-serif`
    - `--font-weight-light`
    - `--font-weight-normal`
    - `--font-weight-bold`
    - `--font-weight-semibold`
    - `--typography-body__font-family`
    - `--typography-heading-2xl__font-size--mobile`
    - `--typography-heading-2xl__font-size`
    - `--typography-heading-3xl__font-size--mobile`
    - `--typography-heading-3xl__font-size`
    - `--typography-heading-3xl__font-weight`
    - `--typography-heading-4xl__font-size--mobile`
    - `--typography-heading-4xl__font-size`
    - `--typography-heading-4xl__font-weight`
    - `--typography-heading-5xl__font-size--mobile`
    - `--typography-heading-5xl__font-size--tablet`
    - `--typography-heading-5xl__font-size`
    - `--typography-heading-lg__font-size`
    - `--typography-heading-lg__font-weight`
    - `--typography-heading-md__font-size`
    - `--typography-heading-md__font-weight`
    - `--typography-heading-sm__font-size`
    - `--typography-heading-sm__font-weight`
    - `--typography-heading-xl__font-size`
    - `--typography-heading__font-family`
    - `--typography-heading__font-weight`
- Added the following font variables:
    - `--font-family-body`
    - `--font-family-button`
    - `--font-family-heading`
    - `--font-family-link`
    - `--font-weight-body-lg`
    - `--font-weight-body-md`
    - `--font-weight-body-sm`
    - `--font-weight-button-lg`
    - `--font-weight-button-md`
    - `--font-weight-button-sm`
    - `--font-weight-heading-2xl`
    - `--font-weight-heading-3xl`
    - `--font-weight-heading-4xl`
    - `--font-weight-heading-5xl`
    - `--font-weight-heading-lg`
    - `--font-weight-heading-md`
    - `--font-weight-heading-sm`
    - `--font-weight-heading-xl`
- Added globally inherited font variables:
    - `--global__font-family`
    - `--global__font-size`
    - `--global__line-height`
- Removed shadow variables:
    - `--shadow-base-blur-radius`
    - `--shadow-base-color`
    - `--shadow-base-offset-x`
    - `--shadow-base-offset-y`
    - `--shadow-base`
    - `--shadow-focus-inverse`
    - `--shadow-focus-link`
    - `--shadow-focus`
- Removed the `ds-u-font-weight--semibold` utility class

### BREAKING changes for healthcare
- Removed in healthcare theme:
    - `--inset__border-width`
    - `--footer__list-marker-size`

### BREAKING changes for medicare
- **Hint text is now larger** and matches the size of inline-error text
- Removed `--hint__font-size` from medicare theme
- Replaced `--choice__translateY` with `--choice-label__top-offset`

### Progress on reconciling CSS and Figma

Here are the checklist items that this batch addresses:

- [ ] Addition (or failure of removal) of `--autocomplete*` variables
- [ ] Change to CMS.gov badge radii
- [x] Missing `--button__font-weight` in Figma
- [ ] Figure out how to use `--dropdown-option-group__padding` and `--dropdown__background-color` in the CSS
- [x] Address all font family and font weight variable changes
- [x] Use new `--global__color*` variables in the reset file
- [x] Find out why `--label__*` variables disappeared
- [x] Stop using `--shadow-base` in CSS
- [ ] Investigate missing and renamed `--usa-banner` variables
- [ ] Figure out an alternative to the `--form*` variables that were removed
- [x] Mark that `--footer__list-marker-size` and `--inset__border-width` were removed from the healthcare theme
- [x] Address changes to choice
- [x] `--hint__font-size` is missing in Figma and only existed in medicare before (set to `inherit` otherwise)

Need to update this PR description with the list of breaking changes.

## How to test

Use Storybook to verify:
- Global text styles are applied to the body element
- Buttons have the right font weight
- Other typographical styles still work correctly
- Choices look correct

You can also run verify that none of the removed variables are still in use in the CSS.
